code/modules/mob/living/carbon/alien/organs.dm

```diff
@@ -167,3 +167,144 @@
+			qdel(thing)
+
+/obj/item/organ/internal/stomach/alien/proc/consume_thing(atom/movable/thing)
+	RegisterSignal(thing, COMSIG_MOVABLE_MOVED, .proc/content_moved)
+	if(isliving(thing))
+		RegisterSignal(thing, COMSIG_LIVING_DEATH, .proc/content_died)
+	stomach_contents += thing
+	thing.forceMove(owner || src) // We assert that if we have no owner, we will not be nullspaced
+
+/obj/item/organ/internal/stomach/alien/proc/content_died(atom/movable/source)
+	SIGNAL_HANDLER
+	qdel(source)
+
+/obj/item/organ/internal/stomach/alien/proc/content_moved(atom/movable/source)
+	SIGNAL_HANDLER
+	if(source.loc == src || source.loc == owner) // not in us? out da list then
+		return
+	stomach_contents -= source
+	UnregisterSignal(source, list(COMSIG_MOVABLE_MOVED, COMSIG_LIVING_DEATH))
+
+/obj/item/organ/internal/stomach/alien/Insert(mob/living/carbon/stomach_owner, special)
+	RegisterSignal(stomach_owner, COMSIG_ATOM_RELAYMOVE, .proc/something_moved)
+	return ..()
+
+/obj/item/organ/internal/stomach/alien/Remove(mob/living/carbon/stomach_owner, special)
+	UnregisterSignal(stomach_owner, COMSIG_ATOM_RELAYMOVE)
+	return ..()
+
+/obj/item/organ/internal/stomach/alien/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	. = ..()
+	if(loc == null && owner)
+		for(var/atom/movable/thing as anything in contents)
+			thing.forceMove(owner)
+	else if(loc != null)
+		for(var/atom/movable/thing as anything in contents)
+			thing.forceMove(src)
+
+/obj/item/organ/internal/stomach/alien/proc/something_moved(mob/living/source, mob/living/user, direction)
+	SIGNAL_HANDLER
+	relaymove(user, direction)
+	return COMSIG_BLOCK_RELAYMOVE
+
+/obj/item/organ/internal/stomach/alien/relaymove(mob/living/user, direction)
+	if(!(user in src.stomach_contents))
+		return
+	if(!prob(40))
+		return
+	var/atom/play_from = owner || src
+	var/stomach_text = owner ? "[owner]'s stomach" : "\the [src]"
+	if(prob(25))
+		play_from.audible_message(span_warning("You hear something rumbling inside [stomach_text]..."), \
+			span_warning("You hear something rumbling."), 4,\
+			self_message = span_userdanger("Something is rumbling inside your stomach!"))
+
+	if(user.client)
+		user.client.move_delay = world.time + 2 SECONDS
+
+	var/obj/item/pokie = user.get_active_held_item()
+	if(!pokie?.force)
+		return
+
+	var/impact = rand(round(pokie.force / 4), pokie.force)
+	applyOrganDamage(impact)
+
+	var/damage_ratio = damage / max(maxHealth, 1)
+	if(owner)
+		var/obj/item/bodypart/part = owner.get_bodypart(BODY_ZONE_CHEST)
+		// Brute damage to the mob is less then to the organ, so there's a higher chance of the explosion happening before xeno death
+		part.receive_damage(impact / 2)
+		// We choose the option that's best for the check
+		var/part_dam_ratio = part.brute_dam / max(part.max_damage, 1)
+		if(damage_ratio < part_dam_ratio)
+			damage_ratio = part_dam_ratio
+
+	play_from.visible_message(span_danger("[user] attacks [stomach_text] wall with the [pokie.name]!"), \
```

Comment:
this should have a message to the user or a vfx (maybe something like the animation of the waddle component

Reply:
good damn idea, I'll make it shake the mob.
what do you mean by message to the user? some feedback to the mob who did it? shouldn't they see the message?

Reply:
i dont think visible message is sent to contents? just like how if you hold an item, you need to send visible message on the thing holding the item, and not the item

Reply:
hm

Reply:
I've added better audio feedback, and some visual flair to it